runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/DoFnOp.java

```diff
@@ -317,7 +422,18 @@ public void processSideInputWatermark(Instant watermark, OpEmitter<OutT> emitter
   }
 
   @Override
-  public void processTimer(KeyedTimerData<Void> keyedTimerData) {
+  public void processTimer(KeyedTimerData<Void> keyedTimerData, OpEmitter<OutT> emitter) {
+    // this is internal timer in processing time to check whether a bundle should be closed
+    if (bundleCheckTimerId.equals(keyedTimerData.getTimerData().getTimerId())) {
+      attemptFinishBundle(emitter);
+      scheduleNextBundleCheck();
+      return;
+    }
+
+    checkState(
```

Comment:
Move this check to the translator so it can throw before running.

We might need this in the long run. Please create a Beam ticket for this and add a comment to link the ticket here.